Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceRenderer.swift

```diff
@@ -10,45 +10,84 @@ import UIKit
 
 @available(iOS 13.0, *)
 internal protocol ExperienceRendering: AnyObject {
-    func show(experience: Experience, priority: RenderPriority, published: Bool, completion: ((Result<Void, Error>) -> Void)?)
-    func show(qualifiedExperiences: [Experience], priority: RenderPriority, completion: ((Result<Void, Error>) -> Void)?)
+    func show(experience: Experience,
+              priority: RenderPriority,
+              published: Bool,
+              experiment: Experiment?,
+              completion: ((Result<Void, Error>) -> Void)?)
+    func show(qualifiedExperiences: [Experience],
+              priority: RenderPriority,
+              experiments: [String: Experiment],
+              completion: ((Result<Void, Error>) -> Void)?)
     func show(stepInCurrentExperience stepRef: StepReference, completion: (() -> Void)?)
     func dismissCurrentExperience(markComplete: Bool, completion: ((Result<Void, Error>) -> Void)?)
     func getCurrentExperienceData() -> ExperienceData?
     func getCurrentStepIndex() -> Experience.StepIndex?
 }
 
+internal enum ExperienceRendererError: Error {
+    case experimentControl
+}
+
 @available(iOS 13.0, *)
 internal class ExperienceRenderer: ExperienceRendering {
 
     private let stateMachine: ExperienceStateMachine
     private let analyticsObserver: ExperienceStateMachine.AnalyticsObserver
     private weak var appcues: Appcues?
     private let config: Appcues.Config
+    private let analyticsPublisher: AnalyticsPublishing
```

Comment:
@mmaatttt this adds a new line in our dependency graph, and one crossing from the presentation side to the analytics side.  I don't love it, but didn't have any better ideas to trigger this event at the spot needed here during `show(...)`.

Reply:
removed in latest refactor.